tools/kmod/sof_remove.sh

```diff
@@ -2,206 +2,13 @@
-
-#-------------------------------------------
-# SOF client drivers
-#-------------------------------------------
-remove_module snd_sof_probes
-remove_module snd_sof_ipc_test
-remove_module snd_sof_ipc_flood_test
-remove_module snd_sof_ipc_msg_injector
-remove_module snd_sof_dma_trace
-
-# snd_sof_nocodec dependencies re-ordered
-# in https://github.com/thesofproject/linux/pull/2800
-# TODO: remove || true and the duplicate below
-# when we stop testing old branches.
-remove_module snd_sof_nocodec || true
-
-remove_module snd_sof
-remove_module snd_sof_nocodec
-remove_module snd_sof_utils
-
-#-------------------------------------------
-# Codec drivers
-#-------------------------------------------
-remove_module snd_soc_da7213
-remove_module snd_soc_da7219
-remove_module snd_soc_pcm512x_i2c
-remove_module snd_soc_pcm512x
-
-remove_module snd_soc_rt274
-remove_module snd_soc_rt286
-remove_module snd_soc_rt298
-remove_module snd_soc_rt700
-remove_module snd_soc_rt711
-remove_module snd_soc_rt1308
-remove_module snd_soc_rt1308_sdw
-remove_module snd_soc_rt715
-remove_module snd_soc_rt711_sdca
-remove_module snd_soc_rt1316_sdw
-remove_module snd_soc_rt715_sdca
-remove_module snd_soc_sdw_mockup
-remove_module snd_soc_rt1011
-remove_module snd_soc_rt5640
-remove_module snd_soc_rt5645
-remove_module snd_soc_rt5651
-remove_module snd_soc_rt5660
-remove_module snd_soc_rt5670
-remove_module snd_soc_rt5677
-remove_module snd_soc_rt5677_spi
-remove_module snd_soc_rt5682_sdw
-remove_module snd_soc_rt5682_i2c
-remove_module snd_soc_rt5682
-remove_module snd_soc_rt5682s
-remove_module snd_soc_rl6231
-remove_module snd_soc_rl6347a
-
-remove_module snd_soc_wm8804_i2c
-remove_module snd_soc_wm8804
-
-remove_module snd_soc_es8316
-remove_module snd_soc_es8326
-
-remove_module snd_soc_max98090
-remove_module snd_soc_ts3a227e
-remove_module snd_soc_max98357a
-remove_module snd_soc_max98373_sdw
-remove_module snd_soc_max98373_i2c
-remove_module snd_soc_max98373
-remove_module snd_soc_max98390
-
-remove_module snd_soc_hdac_hda
-remove_module snd_soc_hdac_hdmi
-remove_module snd_hda_codec_hdmi
-remove_module snd_soc_dmic
-
-remove_module snd_hda_codec_realtek
-remove_module snd_hda_codec_generic
-
-#-------------------------------------------
-# Remaining core SOF parts
-#-------------------------------------------
-remove_module snd_soc_acpi
-remove_module snd_hda_ext_core
-
-remove_module snd_intel_dspcfg
-
-remove_module soundwire_intel_init
-remove_module soundwire_intel
-remove_module soundwire_cadence
-remove_module soundwire_generic_allocation
-remove_module regmap_sdw
-remove_module regmap_sdw_mbq
-remove_module soundwire_bus
-remove_module snd_intel_sdw_acpi
-
-remove_module snd_soc_core
-remove_module snd_hda_codec
-remove_module snd_hda_core
-remove_module snd_hwdep
-remove_module snd_compress
-remove_module snd_pcm_dmaengine
-remove_module snd_pcm
-
+TOPDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd)
+
+case $PLATFORM in
```

Comment:
how is PLATFORM set?

Reply:
"This depends on PR: https://github.com/thesofproject/sof-test/pull/926"

Reply:
@marc-hb very good suggestion. One single comment about:

```
 for remove_sub_script in "$TOPDIR"/tools/kmod/sof_remove_*.sh; do
         "$remove_sub_script"
    done
```

I'm not sure this is the correct way to run the ```sof_remove* scripts``` as for example it doesn't make any sens to try to remove IMX modules on Intel platform or viceversa.